src/app/Processor/ProcessorEdit/ProcessorEdit.tsx

```diff
@@ -116,6 +121,40 @@ const ProcessorEdit = (props: ProcessorEditProps): JSX.Element => {
     });
   }, [name, processorType, filters, transformation, action, source]);
 
+  const prepareFilters: (filters: EventFilter[]) => Partial<EventFilter>[] = (
+    filters: EventFilter[]
+  ) =>
+    filters
+      .filter(
+        (filter) => filter.type && (filter.value || filter.values) && filter.key
+      )
+      .map((filter) => {
+        if (filter.value && isCommaSeparatedFilterType(filter)) {
+          const multipleValuesMapper = (item: string): number | string =>
+            filter.type === FilterType.NUMBER_IN
+              ? parseFloat(item.trim())
+              : item.trim();
+
+          const multipleValuesPredicate =
+            filter.type === FilterType.NUMBER_IN
+              ? (item: number | string): boolean =>
+                  item !== null && !isNaN(item as number)
+              : String;
+
+          const values = filter.value
+            .split(",")
+            .map(multipleValuesMapper)
+            .filter(multipleValuesPredicate);
+
+          return {
+            key: filter.key,
+            type: filter.type,
+            values,
+          };
+        }
+        return filter;
+      });
```

Comment:
Hello Valentino, would you help me/guide, how to measure coverage of this? It seems to me we miss some unit tests. But I may be wrong, and I will learn at least something new.

Reply:
@jomarko I've added a test. Let me know if it's more clear now.